examples/albert/run_first_peer.py

```diff
@@ -1,51 +1,158 @@
 #!/usr/bin/env python
 
 import time
-import argparse
+import subprocess
 import wandb
+from dataclasses import dataclass, asdict
+from typing import Optional, List
 from whatsmyip.providers import GoogleDnsProvider
 from whatsmyip.ip import get_ip
 
 import hivemind
+import torch
+from torch_optimizer import Lamb
+from transformers import AlbertForPreTraining, AlbertConfig, HfArgumentParser
 from hivemind.utils.logging import get_logger
 
+from run_trainer import DatasetArguments, CollaborationArguments, \
+    get_model, get_optimizer_and_scheduler
+
 
 logger = get_logger(__name__)
 
+
+@dataclass
+class CoordinatorArguments(CollaborationArguments):
+    address: Optional[str] = None  # this machine's network address. Use public IP for global experiments,
+    # local address for private runs
+    refresh_period: float = 30  # coordinator will fetch keys from DHT once in this many seconds
+    wandb_project: Optional[str] = None  # learning curves will be published there
+    save_checkpoint_step_interval: int = 5  # coordinator will load and save state from peers once every that many steps
+    upload_model_as: Optional[str] = None  # coordinator will upload the checkpoint to that HuggingFace repo
+    upload_interval: Optional[float] = None  # coordinator will upload model once in this many seconds
+    # Note: You might want to have several initial peers so that if one dies,
+    # new workers still can join the collaboration via alive initial peers' addresses.
+    # Specify initial_peers argument for that purpose
+
+
+class CheckpointHandler:
+    def __init__(self, coordinator_args: CoordinatorArguments, dataset_args: DatasetArguments, dht: hivemind.DHT):
```

Comment:
Receiving the entire DatasetArguments object only to get a config path is probably overkill